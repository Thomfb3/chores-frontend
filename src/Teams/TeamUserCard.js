import React, { useContext } from "react";
import defaultProfileImage from '../assets/images/default-profile-pic.gif';
import UserContext from "../auth/UserContext";
import Avatar from '@mui/material/Avatar';
import StarIcon from '@mui/icons-material/Star';

function TeamUserCard({ id, isCurrentUser, position, profileImage, username, firstName, isAdmin, currentPoints, allTimePoints }) {
    console.debug("TeamUserCard");
    const { currentUser } = useContext(UserContext);

    const defaultProfilePic = (currentUser.profileImage === "defaultProfile.jpg")
        ? defaultProfileImage
        : currentUser.profileImage;

    const determineOrdinal = (position) => {
        if (+position <= 3) return "top-three";
        let str = position.toString();
        if (str.charAt(str.length - 1) === "1") return "st";
        if (str.charAt(str.length - 1) === "2") return "nd";
        if (str.charAt(str.length - 1) === "3") return "rd";
        return "th";
    };

    const determinePositionStyle = (position) => {
        if (position === 1) return "first";
        if (position === 2) return "second";
        if (position === 3) return "third";
        if (position > 3) {
            return `other-number--${determineOrdinal(position)}`;
        };
    };


    return (
        <div className="TeamUserCard">
            {/* <div className="TeamUserCard__box-shadow">
                <div className="TeamUserCard__other"></div>
                <div className={`TeamUserCard__clipped TeamUserCard__clipped--${determinePositionStyle(position)}`}></div>
            </div> */}
            <div className={`TeamUserCard__container TeamUserCard__container--${determinePositionStyle(position)}`}>
                <div className="TeamUserCard__inner-container TeamUserCard__inner-container--align-center">
                    <div className="TeamUserCard__left">
                    <div className="TeamUserCard__position-container">
                        <div className={`TeamUserCard__position TeamUserCard__${determinePositionStyle(position)}`}>{position}</div>
                    </div>
                    {(isAdmin) &&
                        <StarIcon
                            className="TeamUserCard__star"
                            sx={{
           
                            }}
                        />
                    }
                    <Avatar
                        className="Profile__avatar--component"
                        alt={defaultProfilePic}
                        src={defaultProfilePic}
                        sx={{ width: 65, height: 65 }}
                    >{firstName.charAt(0)}</Avatar>
                        <div className="TeamUserCard__name">
                            <div>
                                <p className="TeamUserCard__firstname">{firstName}</p>
                                <p className="TeamUserCard__username">{username}</p>
                            </div>
                            <div className="TeamUserCard__titles">
                                {isCurrentUser ? <p>{`(Me)`}</p> : ""}
                                {isAdmin ? <p>{`(Team Manager)`}</p> : ""}
                            </div>
                        </div>
                    </div>
                    <div className="TeamUserCard__all-points">
                        <p className={`TeamUserCard__all-points--label`}>All Time Points</p>
                        <p className={`TeamUserCard__all-points--points TeamUserCard__points-${determinePositionStyle(position)}`}>{allTimePoints} pts</p>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default TeamUserCard;