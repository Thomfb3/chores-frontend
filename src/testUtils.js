import React from "react";
import UserContext from "./auth/UserContext";
import ChoreCommentContext from "./comments/ChoreCommentContext";

const demoUser = {
  _id: "userid1",
  username: "testuser1",
  firstName: "testfirst1",
  lastName: "testlast1",
  profileImage: "defaultProfile.jpg",
  allTimePoints: 1000,
  currentPoints: 200,
  role: "admin"
};

const testTeam = {
  _id: "testTeam",
  name: "testTeam"
};

const testTeamUsers = [
  {
    _id: "userid1",
    username: "testuser1",
    firstName: "testfirst1",
    lastName: "testlast1",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "admin"
  },
  {
    _id: "userid2",
    username: "testuser2",
    firstName: "testfirst2",
    lastName: "testlast2",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid3",
    username: "testuser3",
    firstName: "testfirst3",
    lastName: "testlast3",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid4",
    username: "testuser4",
    firstName: "testfirst4",
    lastName: "testlast4",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  }
];

let choreComments = [];

const UserProvider =
  ({ children,
    currentUser = demoUser,
    currentTeam = testTeam,
    currentTeamUsers = testTeamUsers,
    isAdmin = false
  }) => (
      <UserContext.Provider value={{ currentUser, currentTeam, currentTeamUsers }}>
        {children}
      </UserContext.Provider>
    );


const CommentProvider =
  ({ children,
    currentChoreComments = choreComments
  }) => (
      <ChoreCommentContext.Provider value={{ currentChoreComments }}>
        {children}
      </ChoreCommentContext.Provider>
    );

export { UserProvider, CommentProvider };

