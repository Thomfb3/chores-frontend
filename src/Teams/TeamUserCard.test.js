import React from "react";
import { render } from "@testing-library/react";
import TeamUserCard from "./TeamUserCard";
import { UserProvider } from "../testUtils";
import { MemoryRouter } from "react-router";


it("matches snapshot", function () {

    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <TeamUserCard
                    id="rewardid1"
                    key="rewardid1"
                    isCurrentUser={true}
                    position={1}
                    profileImage="defaultPic.jpeg"
                    username="userid1"
                    firstName="tester"
                    isAdmin={true}
                    currentPoints={500}
                    allTimePoints={1500}
                />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});
