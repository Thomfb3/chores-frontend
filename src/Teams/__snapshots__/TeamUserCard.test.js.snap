// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TeamUserCard"
  >
    <div
      class="TeamUserCard__container"
    >
      <div
        class="TeamUserCard__inner-container TeamUserCard__inner-container--align-center"
      >
        <div
          class="TeamUserCard__position-container"
        >
          <div
            class="TeamUserCard__position TeamUserCard__first"
          >
            1
          </div>
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-52v5ae-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <div
          class="MuiAvatar-root MuiAvatar-circular Profile__avatar--component css-6w89ng-MuiAvatar-root"
        >
          <img
            alt="default-profile-pic.gif"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="default-profile-pic.gif"
          />
        </div>
        <div
          class="TeamUserCard__name-grow"
        >
          <div>
            <p
              class="TeamUserCard__firstname"
            >
              tester
            </p>
            <p
              class="TeamUserCard__username"
            >
              userid1
            </p>
          </div>
          <div
            class="TeamUserCard__titles"
          >
            <p>
              (Me)
            </p>
            <p>
              (Team Manager)
            </p>
          </div>
        </div>
        <div
          class="TeamUserCard__all-points"
        >
          <p
            class="TeamUserCard__all-points--label TeamUserCard__points-first"
          >
            All Time Points
          </p>
          <p
            class="TeamUserCard__all-points--points TeamUserCard__points-first"
          >
            1500 pts
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
